modules/python/cython/field.pyx

```diff
@@ -0,0 +1,240 @@
+#cython: language_level=3, c_string_type=unicode, c_string_encoding=ascii
+
+cimport field
+from cpython.mem cimport PyMem_Malloc, PyMem_Free, PyMem_Realloc
+from cpython.pycapsule cimport PyCapsule_New, PyCapsule_IsValid, PyCapsule_GetPointer
+from error cimport nitf_Error
+
+import contextlib
+import numpy as np
+from deprecated import deprecated
+from .error import NitfError
+
+
+cpdef enum FieldType:
+    NITF_BCS_A
+    NITF_BCS_N
+    NITF_BINARY
+
+
+cpdef enum ConvType:
+    NITF_CONV_UINT
+    NITF_CONV_INT
+    NITF_CONV_REAL
+    NITF_CONV_STRING
+    NITF_CONV_RAW
+
+
+@contextlib.contextmanager
+def printoptions(*args, **kwargs):
+    """Useful context manager for temporarily changing numpy print options"""
+    original = np.get_printoptions()
+    np.set_printoptions(*args, **kwargs)
+    try:
+        yield
+    finally:
+        np.set_printoptions(**original)
+
+
+cdef class Field:
+    cdef field.nitf_Field* _c_field
+    cdef Py_ssize_t _shape[1]
+    cdef Py_ssize_t _strides[1]
+    cdef void* _buf_value
+    cdef int _ref_count
+
+    def __cinit__(self, capsule=None, value=None, length=None, field_type=FieldType.NITF_BCS_A):
+        cdef nitf_Error error
+
+        self._c_field = NULL
+        self._buf_value = NULL
+        self._strides[0] = 1
+        self._shape[0] = 0
+        self._ref_count = 0
+
+        if capsule is not None:
+            if not PyCapsule_IsValid(capsule, "Field"):
+                raise TypeError("Invalid C pointer type")
+            self._c_field = <field.nitf_Field*>PyCapsule_GetPointer(capsule, "Field")
+        elif length is not None or value is not None:
+            if length is None:
+                length = len(value)
+            self._c_field = nitf_Field_construct(length, field_type, &error)
+            if self._c_field is NULL:
+                raise NitfError(error)
+            if value is not None:
```

Comment:
I'm concerned that a policy of making the caller decide whether they have to manually set the type could lead to annoying bugs if someone forgets. But then it looks like the only cases where something could go wrong are with file length and "Date and Time" fields. Could there be some warning in the .pxd file (I take it that's like a .h for cython?) about these particular cases being manual?